src/crimson/osd/osd_operations/peering_event.cc

```diff
@@ -89,9 +89,12 @@ seastar::future<> PeeringEvent<T>::with_pg(
       return this->template enter_stage<interruptor>(
 	BackfillRecovery::bp(*pg).process);
     }).then_interruptible([this, pg, &shard_services] {
-      pg->do_peering_event(evt, ctx);
-      that()->get_handle().exit();
-      return complete_rctx(shard_services, pg);
+      return interruptor::async([this, pg] {
+	pg->do_peering_event(evt, ctx);
```

Comment:
Are you able to elaborate in comments that where is the dependencies that call `seastar::future::wait()` in the `do_peering_event()` and `handle_advance_map()` implemetations?

I think it's error-prone and thus important to be explicit that certain methods must be wrapped by `async()` when want to call it.

Reply:
Yeah, will do:-)

Reply:
@xxhdx1985126 If I'm interpreting this correctly, the idea is to all PeeringState callbacks to freely run IO.  Reasonable, but I want you to invoke ::async right at the call into the PeeringState, not in the events.  do_peering_event and handle_advance_map should be doing the wrapping as close as possible to the PeeringState state machine invocation, and you need to add comments explaining very obviously why.

Reply:
Indeed, if you can find a single place to do the wrapping, that would be ideal.

Reply:
Yes, will do

Reply:
done, please take a look again